src/main/java/com/Elessar/database/MongoDB.java

```diff
@@ -46,17 +71,35 @@ public void insert(User user) {
 
     }
 
+    @Override
+    public List<Message> findAndUpdate(Message filter, Message update) {
+        final List<Message> findResult = find(filter);
```

Comment:
probably don't need this query

Reply:
I think this is still needed in LogOnHandler

Reply:
probably no longer need this after we fix the logon handler